docs/source/agent.rst

```diff
@@ -25,33 +25,35 @@ In order to satisfy above requirements, we opted for a simple thread-based appro
 all software components that could potentially block into their own threads. In particular, these
 are ``ContractEventMonitor`` and ``EventProcessor``.
 
+With the current implementation, we have a ``EventProcessor`` for every direction in their own thread running for the agent.
+This helps to scale well, as if there is a new L2 chain added, then agent would just create another directions which has the new
+chain and there is no need to run seperate processes for this manner. Also, we have L2 chain based ``ContractEventMonitors`` that 
+event processors can subscribe and get events from. This prevents us to create redundant event monitors for event processors.
+
 
 ContractEventMonitor
 --------------------
 
-``ContractEventMonitor`` listens for blockchain events emitted by a given contract, decodes the events
+``ContractEventMonitor`` listens for blockchain events emitted by given contracts, decodes the events
 into an internal event representation (event types in ``beamer.events``) and forwards decoded events
 to the ``EventProcessor``. The event monitor does not query the JSON-RPC server for events directly.
 Rather, it does that via an instance of ``EventFetcher``. Event fetcher handles communication with the
-JSON-RPC server and can block. However, even if the event fetcher for, say, L2a chain, blocks, since
+JSON-RPC server and can block. However, even if the event fetcher for, say, source chain, blocks, since
 it is only being invoked by the contract event monitor, which runs inside a thread, it won't block
 the entire agent. Therefore, even if a JSON-RPC server is very slow or the connection is otherwise
 unreliable, the agent as a whole will remain responsive.
 
-With the current contract implementation, we have one contract, ``RequestManager``, deployed on the
-source chain (L2a), and one contract, ``FillManager``, deployed on the target chain (L2b). Therefore
-we have two event monitors, one for each contract, and each event monitor has its own event fetcher,
-as can be seen in the figure above.  Each pair ``(EventFetcher, ContractEventMonitor)`` works
+With the current contract implementation, we have both ``RequestManager`` and ``FillManager`` for L2 chain.
```

Comment:
This sentence looks weird, I am not sure I understand what we want to say here.

Reply:
Trying to say our eventfetcher now looking both. RequestManager and FillManager :D